common/lib/capa/capa/util.py

```diff
@@ -100,20 +100,27 @@ def contextualize_text(text, context):  # private
     Takes a string with variables. E.g. $a+$b.
     Does a substitution of those variables from the context
     """
+    def convert_to_str(value):
+        """The method tries to convert unicode/non-ascii values into string"""
+        try:
+            return str(value)
+        except UnicodeEncodeError:
```

Comment:
I don't think you need this exception clause.  `value` here will always be a string already, so `str(value)` is basically a no-op.  It won't raise this exception.  If you have a scenario that does raise this exception, it should go into the test cases.

Reply:
@nedbat When we pass non-ascii data from the test (`context = {'a': u'あなたあなたあなた', 'b': u'あなたhi'}`), where text to look for is `text= u'$あなた$a $b'`; this method gets `u'$あなた$a $b'` as `value` which would fail when the method will try `str(value)`. 

Whenever we have a unicode value `str(value)` will raise `UnicodeEncodeError`. 

I think the case is already in tests? I hope my comment was clear.